ui/lib/CancelStream.tsx

```diff
@@ -1,7 +1,86 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
+import { Stream } from "../../sdk/js/index";
+import { formatTime } from "./utils";
 
-const CancelStream = () => {
-  return <div />;
-};
+interface CancelStreamProps {
+  stream: Stream;
+  onComplete: (result: string) => void;
+}
+
+const CancelStream: React.FC<CancelStreamProps> = (props) => {
+  const [result, setResult] = useState<string | null>(null);
+  // Allow user to cancel stream only if he didnt already click on cancel  AND if the stream is cancellable
+  const [isButtonEnabled, setButtonEnabled] = useState(false);
+  // Get the minimum stream life, before which a stream cannot be Canceled
+  const minStreamLife = props.stream.streamManager.MIN_STREAM_LIFE;
+
+  // Check if the stream is cancellable and set the button state accordingly.
+  const checkStreamCancelable = async () => {
+    try {
+      const isCancelable = await props.stream.isCancelable();
```

Comment:
I wonder if we can make this a sync function now that the min stream life is available